views/js/previewer/provider/item/item.js

```diff
@@ -217,6 +217,41 @@ define([
                 .on('flush', () => {
                     this.destroy();
                 });
+        /**
+         * Convenience function to load the current item from the testMap
+         * @returns {Object?} the current item if any or falsy
+         */
+            this.getCurrentItem = function getCurrentItem() {
```

Comment:
DRY
```javascript
this.getFromContext = (key)=>{
  const testContext = this.getTestContext();
  const testMap     = this.getTestMap();

  if (testContext && testMap && testContext[key]) {
     return mapHelper.getItem(testMap, testContext[key]);
  }
}

this.getCurrentItem = () => this.getFromContext('itemIdentifier');
this.getCurrentSection = () => this.getCurrentSection('sectionId');
this.getCurrentPart = () => this.getCurrentSection('testPartId');
```

Reply:
![image](https://user-images.githubusercontent.com/34692207/87679420-702b9d00-c77c-11ea-946d-c312f2c04d46.png)

In this case, refactor this, needs to refactor in `npm package` that affects currentGen runner and need to test by QA everything then. Little out of scope.

I can proposal you create a debt tech issue in this case

Reply:
@habemuscode you can improve code here, and for npm package, please, create ticket

Reply:
https://oat-sa.atlassian.net/browse/TBD-50

Reply:
please, improve code here

Reply:
Done

Reply:
I can't improve this code before improve in npm package before as I said

Reply:
done